### PR TITLE
feat(apm): add apm transactions on requests.

### LINKED ISSUE #9: add apm on transaction
Add APM transaction on all the client requests.
Also add a span for the kartoffel request.

### PR DESCRIPTION
**Pull request by yonatandt**
closes #9
Signed-off-by: YonatanTal <yonatald@gmail.com>

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
 
-import * as logger from 'morgan';
+import * as apm from 'elastic-apm-node';
+import { once } from 'events';
 import * as express from 'express';
-import * as http from 'http';
 import * as helmet from 'helmet';
-import { once } from 'events';
-import { ResourceNotFoundError } from './utils/errors/client.error';
+import * as http from 'http';
```
Comment (lustigy96) [inline src/server.ts:6]: We said that this http is וnnecessary

File: src/utils/wrappers.ts [hunk 2]
```diff
@@ -1,22 +1,74 @@
...
-import { ApplicationError } from './errors/application.error';
+import * as apm from 'elastic-apm-node';
+import { NextFunction, Request, Response } from 'express';
+import { apmConfig } from '../config';
+import { ApplicationError, ClientError } from './errors/application.error';
+
+function extractTraceparent(req: Request): string | undefined {
+  const traceparent = req.header(apmConfig.traceParentHeader); // add as config
+  return traceparent;
+}
```
Comment (lustigy96) [inline src/utils/wrappers.ts]:
```suggestion
function extractTraceparent(req: Request): string | undefined => {
   req.header(apmConfig.traceParentHeader); // add as config
}
```
This syntax is better for eslint.

**Review by lustigy96 — CHANGES_REQUESTED** [review]

File: src/server.ts (hunk 1 above)
Comment (yonatandt) [reply src/server.ts:6]: `http` is necessary. We should use the standard library of `http` and not one from npm like [this](https://www.npmjs.com/package/http).

File: src/utils/wrappers.ts (hunk 2 above)
Comment (yonatandt) [reply src/utils/wrappers.ts]: changed the style.

**Review by lustigy96 — APPROVED** [review]

**Pull request merged**